File: tags/release-1.0/java/core/src/org/daisy/urakawa/undo/CannotRedoException.java
package org.daisy.urakawa.undo;

import org.daisy.urakawa.exception.CheckedException;

/**
 * <p>
 * This exception is raised when trying to redo an action, but the operation
 * fails.
 * </p>
 */
public class CannotRedoException extends CheckedException {
	/**
	 * 
	 */
	private static final long serialVersionUID = 3431387950719820251L;
}

File: branches/sdk-core-structedit/exception/CannotRedoException.cs
using System;

namespace urakawa.exception
{
    /// <summary>
    /// Thrown when an operation cannot be redone.
    /// </summary>
    public class CannotRedoException : CheckedException
    {
        /// <summary>
        /// Constructor setting the message of the exception
        /// </summary>
        /// <param name="msg">The message</param>
        public CannotRedoException(string msg)
            : base(msg)
        {
        }

        /// <summary>
        /// Constructor setting the message and inner <see cref="Exception"/> of the exception
        /// </summary>
        /// <param name="msg">The message</param>
        /// <param name="inner">The inner exception</param>
        public CannotRedoException(string msg, Exception inner)
            : base(msg, inner)
        {
        }
    }
}